jablib/src/main/java/org/jabref/logic/quality/consistency/BibliographyConsistencyCheck.java

```diff
@@ -62,64 +82,115 @@ public record EntryTypeResult(Collection<Field> fields, SequencedCollection<BibE
+                    sortedEntries = new ArrayList<>(entries);
+                } else {
+                    sortedEntries = filterEntriesWithFieldDifferences(entries, differingFields, fieldsInAllEntries);
+                }
+            } else {
+                sortedEntries = filterEntriesWithFieldDifferences(entries, differingFields, fieldsInAllEntries);
+            }
+
+            if (!sortedEntries.isEmpty()) {
+                sortedEntries.sort(new FieldComparatorStack<>(List.of(
+                        new BibEntryByCitationKeyComparator(),
+                        new BibEntryByFieldsComparator()
+                )));
+                resultMap.put(entryType, new EntryTypeResult(differingFields, sortedEntries));
+            }
         }
 
         return new Result(resultMap);
     }
 
-    private static void collectEntriesIntoMaps(List<BibEntry> entries, Map<EntryType, Set<Field>> entryTypeToFieldsInAnyEntryMap, Map<EntryType, Set<Field>> entryTypeToFieldsInAllEntriesMap, Map<EntryType, Set<BibEntry>> entryTypeToEntriesMap) {
+    private static void collectEntriesIntoMaps(BibDatabaseContext bibContext, Map<EntryType, Set<Field>> entryTypeToFieldsInAnyEntryMap, Map<EntryType, Set<Field>> entryTypeToFieldsInAllEntriesMap, Map<EntryType, Set<BibEntry>> entryTypeToEntriesMap) {
+        BibDatabaseMode mode = bibContext.getMode();
+        List<BibEntry> entries = bibContext.getEntries();
+
+        Set<EntryType> typeSet = switch (mode) {
+            case BIBLATEX -> BiblatexEntryTypeDefinitions.ALL.stream()
+                    .map(BibEntryType::getType)
+                    .collect(Collectors.toSet());
+            case BIBTEX -> BibtexEntryTypeDefinitions.ALL.stream()
+                    .map(BibEntryType::getType)
+                    .collect(Collectors.toSet());
+            default -> Set.of();
+        };
+
         for (BibEntry entry : entries) {
-            EntryType entryType = entry.getType();
+            if (typeSet.contains(entry.getType())) {
+                EntryType entryType = entry.getType();
 
-            entryTypeToFieldsInAnyEntryMap
-                    .computeIfAbsent(entryType, _ -> new HashSet<>())
-                    .addAll(filterExcludedFields(entry.getFields()));
+                Set<Field> filteredFields = filterExcludedFields(entry.getFields());
 
-            entryTypeToFieldsInAllEntriesMap
-                    .computeIfAbsent(entryType, _ -> new HashSet<>(filterExcludedFields(entry.getFields())))
-                    .retainAll(filterExcludedFields(entry.getFields()));
+                entryTypeToFieldsInAnyEntryMap
+                        .computeIfAbsent(entryType, _ -> new HashSet<>())
+                        .addAll(filteredFields);
 
-            entryTypeToEntriesMap
-                    .computeIfAbsent(entryType, _ -> new HashSet<>())
-                    .add(entry);
+                if (!entryTypeToFieldsInAllEntriesMap.containsKey(entryType)) {
```

Comment:
Swap branches --> negations in if conditions should be avoided

Add a comment why this is the flow - I cannot get it by reading.